src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala

```diff
@@ -0,0 +1,35 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import org.broadinstitute.dsde.rawls.model._
+
+class GraphAuthDAO extends AuthDAO with GraphDAO {
+  override def saveUser(rawlsUser: RawlsUser, txn: RawlsTransaction): RawlsUser = txn withGraph { db =>
+    val vertex = getUserVertex(db, rawlsUser.userEmail).getOrElse(addVertex(db, VertexSchema.User))
+    saveObject[RawlsUser](rawlsUser, vertex, None, db)
+    rawlsUser
+  }
+
+  override def saveGroup(rawlsGroup: RawlsGroup, txn: RawlsTransaction): RawlsGroup = txn withGraph { db =>
+    val vertex = getGroupVertex(db, rawlsGroup.groupName).getOrElse(addVertex(db, VertexSchema.Group))
+    saveObject[RawlsGroup](rawlsGroup, vertex, None, db)
+    rawlsGroup
+  }
+
+  override def saveWorkspaceAccessGroups(workspaceName: WorkspaceName, userEmail: String, txn: RawlsTransaction): Map[String, RawlsGroupRef] = {
+    val user = RawlsUser(userEmail)
+    saveUser(user, txn)
+
+    // add user to Owner group only
+    val oGroup = RawlsGroup(UserAuth.toWorkspaceAccessGroupName(workspaceName, WorkspaceAccessLevel.Owner), Set(user), Set.empty)
+    val wGroup = RawlsGroup(UserAuth.toWorkspaceAccessGroupName(workspaceName, WorkspaceAccessLevel.Write), Set.empty, Set.empty)
+    val rGroup = RawlsGroup(UserAuth.toWorkspaceAccessGroupName(workspaceName, WorkspaceAccessLevel.Read), Set.empty, Set.empty)
+    saveGroup(oGroup, txn)
+    saveGroup(wGroup, txn)
+    saveGroup(rGroup, txn)
```

Comment:
Maybe `createWorkspaceAccessGroups` would be a better name, because you're right about the purpose.  It could call a specialized `createGroup` which will not allow overwriting.